### ISSUE TITLE
validation of namespaces could be more specific?

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida_core#1703]
@muhrin @sphuber We get
`ValueError: required value was not provided for 'code'`
as an error, but it's not clear if it comes deep down in the sub-workflows (expose_inputs).
Would it be possible to tell more information about which process and which namespace, for instance?

**Comment by giovannipizzi** [comment]
Similarly, we could try to be more specific here:
`ValueError: Invalid type <type 'dict'> for dynamic port value: expected (<class 'aiida.orm.data.Data'>, <class 'aiida.orm.implementation.django.calculation.Calculation'>)`

**Comment by DanielMarchand** [comment]
I just encountered this issue, definitely would be nice to get a more specific error :)

**Comment by muhrin** [comment]
Thanks for the input @DanielMarchand .  I'll evaluate the work needed to get this done this week.

**Comment by sphuber** [comment]
> Similarly, we could try to be more specific here:
> `ValueError: Invalid type <type 'dict'> for dynamic port value: expected (<class 'aiida.orm.data.Data'>, <class 'aiida.orm.implementation.django.calculation.Calculation'>)`

This has been reported as a separate issue in #1980

**Comment by sphuber** [comment]
This is now fixed in `plumpy==0.12.1` and the dependency has been updated in PR #2324 
Error message is now for example:
```
ValueError: Error occurred validating port 'inputs.a': required value was not provided for 'a'
```
